### PR TITLE
[ci] do not mock Authenticator class in kerberos request spec

### PR DESCRIPTION
**Pull request by evanrolfe**

### CODE REVIEW

File: src/api/spec/requests/kerberos_login_spec.rb [hunk 1]
```diff
@@ -24,37 +24,28 @@
...
-            and_raise(Authenticator::AuthenticationRequiredError, 'something happened')
+          get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate INVALID_TICKET" }
         end
 
-        it 'does not authenticate the user' do
-          get "/source.xml", headers: { 'X-HTTP_AUTHORIZATION' => auth_header }
-          expect(response).to have_http_status(:unauthorized)
-          expect(response.body).to match('something happened')
-        end
+        it { expect(response).to have_http_status(:unauthorized) }
```
Comment (bgeuken) [inline src/api/spec/requests/kerberos_login_spec.rb:47]: There is nothing that ensures that users will the error, right?

**Comment by bgeuken** [comment]
Looks like you dropped the test that ensures that OBS starts the negotiation mode

**Comment by evanrolfe** [comment]
> Looks like you dropped the test that ensures that OBS starts the negotiation mode

Do you mean [this](https://github.com/evanrolfe/open-build-service/blob/bb76e95bbb076ef6f302acb425801a83b35fe414/src/api/spec/requests/kerberos_login_spec.rb#L17-L24) spec? Its still there?

File: src/api/spec/requests/kerberos_login_spec.rb (hunk 1 above)
Comment (evanrolfe) [reply src/api/spec/requests/kerberos_login_spec.rb:47]: What do you mean?

Comment (evanrolfe) [reply src/api/spec/requests/kerberos_login_spec.rb:47]: I can add this line if thats what you're talking about?

```ruby
it { expect(response.body).to match('Received invalid GSSAPI context')}
```

Comment (bgeuken) [reply src/api/spec/requests/kerberos_login_spec.rb:47]: All we know here is that the server responds with a 401 here or am I missing something?

**Comment by bgeuken** [comment]
I see. I've overlooked it

**Pull request merged**